iio/iio_axi_adc/iio_axi_adc.c

```diff
@@ -0,0 +1,548 @@
+	.store = set_calibbias,
+};
+static struct iio_attribute iio_attr_calibscale = {
+	.name = "calibscale",
+	.show = get_calibscale,
+	.store = set_calibscale,
+};
+static struct iio_attribute iio_attr_samples_pps = {
+	.name = "samples_pps",
+	.show = get_samples_pps,
+	.store = set_samples_pps,
+};
+static struct iio_attribute iio_attr_sampling_frequency = {
+	.name = "sampling_frequency",
+	.show = get_sampling_frequency,
+	.store = set_sampling_frequency,
+};
+static struct iio_attribute *iio_voltage_attributes[] = {
+	&iio_attr_calibphase,
+	&iio_attr_calibbias,
+	&iio_attr_calibscale,
+	&iio_attr_samples_pps,
+	&iio_attr_sampling_frequency,
+	NULL,
+};
+static struct iio_channel iio_channel_voltage0 = {
+	.name = "voltage0",
+	.attributes = iio_voltage_attributes,
+};
+static struct iio_channel iio_channel_voltage1 = {
+	.name = "voltage1",
+	.attributes = iio_voltage_attributes,
+};
+static struct iio_channel iio_channel_voltage2 = {
+	.name = "voltage2",
+	.attributes = iio_voltage_attributes,
+};
+static struct iio_channel iio_channel_voltage3 = {
+	.name = "voltage3",
+	.attributes = iio_voltage_attributes,
+};
+
+static struct iio_channel *iio_adc_channels[] = {
+	&iio_channel_voltage0,
+	&iio_channel_voltage1,
+	&iio_channel_voltage2,
+	&iio_channel_voltage3,
+	NULL,
+};
+
+/**
+ * @brief Get an axi_adc xml.
+ * @param xml - xml string.
+ * @param device_name.
+ * @param ch_no.
+ * @return SUCCESS in case of success or negative value otherwise.
+ */
+ssize_t iio_axi_adc_get_xml(char** xml, const char *device_name, uint8_t ch_no)
+{
+	char buff[256];
+	struct xml_node *device = NULL;
+	struct xml_node *channel = NULL;
+	struct xml_node *attribute = NULL;
+	struct xml_attribute *att = NULL;
+	struct xml_document *document = NULL;
+	ssize_t ret;
+
+	ret = xml_create_node(&device, "device");
+	if (ret < 0)
+		goto error;
+	ret = xml_create_attribute(&att, "id", device_name);
+	if (ret < 0)
+		goto error;
+	ret = xml_add_attribute(device, att);
+	if (ret < 0)
+		goto error;
+	ret = xml_create_attribute(&att, "name", device_name);
+	if (ret < 0)
+		goto error;
+	ret = xml_add_attribute(device, att);
+	if (ret < 0)
+		goto error;
+
+	for (uint8_t i = 0; i < ch_no; i++) {
+		ret = xml_create_node(&channel, "channel");
+		if (ret < 0)
+			goto error;
+		sprintf(buff, "voltage%d", i);
+		ret = xml_create_attribute(&att, "id", buff);
+		if (ret < 0)
+			goto error;
+		ret = xml_add_attribute(channel, att);
+		if (ret < 0)
+			goto error;
+		ret = xml_create_attribute(&att, "type", "input");
+		if (ret < 0)
+			goto error;
+		ret = xml_add_attribute(channel, att);
+		if (ret < 0)
+			goto error;
+
+		ret = xml_create_node(&attribute, "scan-element");
+		if (ret < 0)
+			goto error;
+		sprintf(buff, "%d", i);
+		ret = xml_create_attribute(&att, "index", buff);
+		if (ret < 0)
+			goto error;
+		ret = xml_add_attribute(attribute, att);
+		if (ret < 0)
+			goto error;
+		ret = xml_create_attribute(&att, "format", "le:S12/16&gt;&gt;0");
```

Comment:
This information should not be hard coded. I guess that we should have at least a resolution member in the struct axi_adc. However, since we don't have it yet, let's set this to S16/16 - the correct setting for ADRV9371 and ADRV9009.